addons/account_tax_python/tests/test_tax.py

```diff
@@ -6,56 +6,85 @@
 @tagged('post_install', '-at_install')
 class TestTaxPython(TestTaxCommon):
 
-    @classmethod
-    def setUpClass(cls):
-        super(TestTaxPython, cls).setUpClass()
-        cls.python_tax = cls.env['account.tax'].create({
-            'name': 'Python TAx',
+    def python_tax(self, python_compute, **kwargs):
+        self.number += 1
+        return self.env['account.tax'].create({
+            **kwargs,
+            'name': f"code_({self.number})",
             'amount_type': 'code',
             'amount': 0.0,
-            'python_compute': 'result = ((price_unit * quantity) - ((price_unit * quantity) / 1.12)) * 0.5',
-            'sequence': 1,
+            'python_compute': python_compute,
         })
 
-    def test_tax_python_basic(self):
-        res = self.python_tax.compute_all(130.0)
-        self._check_compute_all_results(
-            136.96, # 'total_included'
-            130.0,  # 'total_excluded'
-            [
-                # base , amount    | seq | amount | incl | incl_base
-                # --------------------------------------------------
-                (130.0, 6.96),   # |  1  |    6%  |   t  |
-                # --------------------------------------------------
-            ],
-            res
+    def test_python_taxes_for_l10n_in(self):
+        tax1 = self.python_tax("result = max(quantity * price_unit * 0.21, quantity * 4.17)")
+        self._check_tax_results(
+            tax1,
+            {
+                'total_included': 157.3,
+                'total_excluded': 130.0,
+                'tax_values_list': (
+                    (130.0, 27.3),
+                ),
+            },
+            130.0,
```

Comment:
why changing the previous test cases? how can I know the behavior didn't change :weary:

Reply:
sorry :see_no_evil: A lot of tests were duplicated or testing the same things and all the time partially. I made all of them checking both rounding computations but all of them are still there